### PR TITLE
[wasi] Fix WasmLinkIcalls and WasmNativeStrip support.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/wasi/build/WasiApp.Native.targets
Comment (campersau) [inline src/mono/wasi/build/WasiApp.Native.targets:178]: Swap:
```suggestion
      <_WasmCommonCFlags Condition="'$(WasmLinkIcalls)' == 'true'"         Include="-DLINK_ICALLS=1" />
```

**Comment by vargaz** [comment]
Looks like the aot compiler is not included in the wasi runtime pack.

**Comment by lewing** [comment]
I thought AOT didn't currently work on wasi due to the inability to rely on runtime features?

**Comment by vargaz** [comment]
The WasmLinkIcalls task relies on it to generate the icall table in the runtime.

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.json.in
Comment (campersau) [inline src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.json.in:31]:
```suggestion
        "Microsoft.NET.Runtime.WebAssembly.Templates",
```

**Comment by vargaz** [comment]
Depends on:
https://github.com/dotnet/runtime/pull/92095

**Review by lewing — APPROVED** [review]
Blocked on https://github.com/dotnet/runtime/pull/92095 but looks good once that lands

**Comment by lewing** [comment]
Build analysis is green

**Pull request merged**
